crates/block-producer/src/runner.rs

```diff
@@ -159,6 +160,23 @@ async fn poll_loop(
             // update tip
             tip_number = raw_header.number().unpack();
             tip_hash = block.header().hash().into();
+
+            // update global hardfork info
+            let hardfork_switch = rpc_client.get_hardfork_switch().await?;
+            let rpc32_epoch_number = hardfork_switch.rfc_0032();
+            *GLOBAL_HARDFORK_SWITCH.lock().await = hardfork_switch;
```

Comment:
Skip the updating if value isn't changed

Reply:
updated